pkg/genericNotes/mocks/GenericNoteService_test.go

```diff
@@ -0,0 +1,244 @@
+
+		mockedNoteRepo.On("FindByIdentifier", mock.AnythingOfType("int"), mock.AnythingOfType("repository.NoteType")).Return(&repository3.GenericNote{Id: 1}, nil)
+		req := &repository3.GenericNote{}
+		resp, err := genericNoteSvc.Save(tx, req, userId1)
+		testErrorAssertions(tt, resp, clusterNoteNotFoundError, err)
+	})
+
+	t.Run("Test Error Case, error in saving genericNote entry", func(tt *testing.T) {
+		genericNoteSvc, mockedNoteRepo, _, _ := initGenericNoteService(tt)
+		tx := &pg.Tx{}
+		testErr := "GenericNote Save error"
+		mockedNoteRepo.On("FindByIdentifier", mock.AnythingOfType("int"), mock.AnythingOfType("repository.NoteType")).Return(&repository3.GenericNote{}, nil)
+		mockedNoteRepo.On("Save", tx, mock.AnythingOfType("*repository.GenericNote")).Return(errors.New(testErr))
+		req := &repository3.GenericNote{}
+		resp, err := genericNoteSvc.Save(tx, req, userId1)
+		testErrorAssertions(tt, resp, testErr, err)
+	})
+
+	t.Run("Test Error Case, error in saving genericNote history audit", func(tt *testing.T) {
+		genericNoteSvc, mockedNoteRepo, mockedHistorySvc, _ := initGenericNoteService(tt)
+		tx := &pg.Tx{}
+		req := &repository3.GenericNote{}
+		testErr := "saving genericNote history audit error"
+		mockedNoteRepo.On("FindByIdentifier", mock.AnythingOfType("int"), mock.AnythingOfType("repository.NoteType")).Return(&repository3.GenericNote{}, nil)
+		mockedNoteRepo.On("Save", tx, mock.AnythingOfType("*repository.GenericNote")).Return(nil)
+		mockedHistorySvc.On("Save", tx, mock.AnythingOfType("*genericNotes.GenericNoteHistoryBean"), mock.AnythingOfType("int32")).Return(nil, errors.New(testErr))
+		resp, err := genericNoteSvc.Save(tx, req, userId1)
+		testErrorAssertions(tt, resp, testErr, err)
+	})
+
+	t.Run("Test Error Case, error in getting user by Id", func(tt *testing.T) {
+		genericNoteSvc, mockedNoteRepo, mockedHistorySvc, mockedUserRepo := initGenericNoteService(t)
+		tx := &pg.Tx{}
+		mockedNoteRepo.On("FindByIdentifier", mock.AnythingOfType("int"), mock.AnythingOfType("repository.NoteType")).Return(&repository3.GenericNote{}, nil)
+		mockedNoteRepo.On("Save", tx, mock.AnythingOfType("*repository.GenericNote")).Return(nil)
+		mockedHistorySvc.On("Save", tx, mock.AnythingOfType("*genericNotes.GenericNoteHistoryBean"), mock.AnythingOfType("int32")).Return(nil, nil)
+		req := &repository3.GenericNote{
+			Description: "test description",
+			AuditLog: sql.AuditLog{
+				UpdatedBy: userId1,
+			},
+		}
+		testErr := "GetUserById error"
+		mockedUserRepo.On("GetById", req.UpdatedBy).Return(nil, errors.New(testErr))
+		resp, err := genericNoteSvc.Save(tx, req, userId1)
+		testErrorAssertions(tt, resp, testErr, err)
+	})
+
+	t.Run("Test Success Case", func(tt *testing.T) {
+		genericNoteSvc, mockedNoteRepo, mockedHistorySvc, mockedUserRepo := initGenericNoteService(t)
+		tx := &pg.Tx{}
+		mockedNoteRepo.On("FindByIdentifier", mock.AnythingOfType("int"), mock.AnythingOfType("repository.NoteType")).Return(&repository3.GenericNote{}, nil)
```

Comment:
why aren't we using actual values in params ??

Reply:
fixed